File: src/client/java/fr/hardel/asset_editor/client/javafx/lib/data/StudioElementId.java
package fr.hardel.asset_editor.client.javafx.lib.data;

import net.minecraft.resources.Identifier;

public record StudioElementId(Identifier identifier, String registry) {

    public static StudioElementId parse(String raw) {
        if (raw == null || raw.isBlank())
            return null;
        String[] parts = raw.split("\\|", 2);
        String idPart = parts[0];
        String registryPart = parts.length > 1 ? parts[1] : null;
        Identifier identifier = Identifier.tryParse(idPart);
        if (identifier == null)
            return null;
        return new StudioElementId(identifier, registryPart == null || registryPart.isBlank() ? null : registryPart);
    }

    public String namespace() {
        return identifier.getNamespace();
    }

    public String resourcePath() {
        return identifier.getPath();
    }
}

File: src/client/kotlin/fr/hardel/asset_editor/client/compose/lib/data/StudioElementId.kt
package fr.hardel.asset_editor.client.compose.lib.data

import net.minecraft.resources.Identifier

data class StudioElementId(val identifier: Identifier, val registry: String?) {

    fun namespace(): String = identifier.namespace

    fun resourcePath(): String = identifier.path

    companion object {
        @JvmStatic
        fun parse(raw: String?): StudioElementId? {
            if (raw.isNullOrBlank()) {
                return null
            }

            val parts = raw.split("|", limit = 2)
            val idPart = parts[0]
            val registryPart = if (parts.size == 2) parts[1] else null
            val identifier = Identifier.tryParse(idPart) ?: return null
            return StudioElementId(identifier, registryPart?.takeUnless { it.isBlank() })
        }
    }
}
